torch/distributions/kl.py

```diff
@@ -207,6 +208,24 @@ def _kl_exponential_exponential(p, q):
     return t1 + rate_ratio - 1
 
 
+@register_kl(ExponentialFamily, ExponentialFamily)
+def _kl_expfamily_expfamily(p, q):
+    if not type(p) == type(q):
+        raise ValueError("The cross KL-divergence between different exponential families cannot \
```

Comment:
I believe this should be a `NotImplementedError`: it is a shortcoming of our implementation rather than a user error due to poorly constructed data.